### PR TITLE
Fix building with Xcode 10

### PR DESCRIPTION
**Pull request by hrubymar10**
# Problem description
Whole problem is described in issue #3350 

# TODO
- [x] Rename problematic files - 49f6736798842c4613e2ab8a7c7c1dfafa734422
- [x] Fix Visual Studio project - ac43a2d3ed9b855f932278a6f88cab1cffae20e1
- [x] Fix CMake, Scons and CodeBlocks builds - ef7a24e74b88d6179db0b5f8856d9482e57f203a
- [ ] Fix issue 1) Load screen missing - NOT CAUSED BY XCODE 10
- [x] Fix issue 2) Blurry video in window mode - 5a89822caa93481f37a9d4b61f524d34f9932ed0
- [x] Fix issue 3) Bad resolution in fullscreen - 5a89822caa93481f37a9d4b61f524d34f9932ed0

# New bugs introduced by XCode 10
## 1) Blank screen on start
Game starts with blank screen. You can click elements (you can hear click sound) but you won't see anything until you move window. When you move window or resize it video appears. This issue is not caused by Xcode 10 but by macOS 10.14 api changes. I'll create new issue for tracking macOS 10.14 specific problems.
image

## 2) Blurry video in window mode
2018-08-11 at 14 33

## 3) Bad resolution in fullscreen
When I switch game to fullscreen display starts to be cropped. Click coordinates works as expected (means that when you click somewhere in bottom right side, you will got window with credits or game quits or preferences appears (depends on where exactly you click)
image

(Before click)
2018-08-11 at 14 27

(After click)
image

### CODE REVIEW

**Comment by hrubymar10** [comment]
**Update**
Issue 2) and 3) are probably caused by retina/Hi-DPi scalling on my MacBook Pro (with retina display)

**Bonus**
Wesnoth now supports darkmode on macOS 10.14 😄
_what is totally important..._

2018-08-11 at 14 37

**Comment by hrubymar10** [comment]
@Vultraz , can I include this flag to all builds (including windows, linux, ...) or only for apple? (`#if defined(__APPLE__)`).

Also another question: It is easy fix but isn't it just workaround?

**Comment by Vultraz** [comment]
That flag is indeed meant to be used on high-DPI systems, though I think it's really meant for macOS, since Windows doesn't use the macOS method of having a canvas 2x the size rendered in 1x the space. Should be fine on all builds.

File: src/video.cpp [hunk 1]
```diff
@@ -232,6 +232,7 @@ void CVideo::init_window()
 
 	// Add any more default flags here
 	window_flags |= SDL_WINDOW_RESIZABLE;
+    window_flags |= SDL_WINDOW_ALLOW_HIGHDPI;
```
Comment (jyrkive) [inline src/video.cpp]: This line should be indented with tabs.

Comment (hrubymar10) [reply src/video.cpp]: F**k... Again? 😃

Comment (hrubymar10) [reply src/video.cpp]: Fixed - 00210065632ee7397699e7e9abf36b9822cbd69e

**Comment by hrubymar10** [comment]
**Update**
I just realised that issue 1) is little bit different. It starts with loadscreen but you can't see anything until you move window. Redraw order?

**Comment by hrubymar10** [comment]
Ok, maybe we can merge this PR and create new issue for blank screen.

**Comment by hrubymar10** [comment]
I am on right track with issue number 1

This issue is NOT cause by Xcode 10 but by macOS 10.14 SDK. Please merge this PR as all problems with Xcode 10 are fixed and I'll open new issue for tracking macOS 10.14 specific issues.

**Comment by CelticMinstrel** [comment]
Needs to be done on master too now, right?

**Pull request merged**